src/extension.node.ts

```diff
@@ -243,9 +243,14 @@ function addConsoleLogger() {
     }
 }
 
+function escapeRegExp(text: string) {
+    return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+}
+
 function tryGetUsername() {
     try {
-        return userInfo().username;
+        const username = escapeRegExp(userInfo().username);
+        return new RegExp(username, 'ig');
```

Comment:
This seems pretty fragile, can we use indexOf/contains instead?

Reply:
to escape each occurrence of a regex special character? That sounds more fragile

Reply:
I mean do plain string comparisons without touching regex at all. If you replace one of these special characters isn't this logic going to fall over? Posix paths can have many more characters than Windows paths for example which could be part of `$HOME`.

Reply:
I think it would still work after escaping the character, I'm not removing them.
But yeah, I can try to manually find and replace all occurrences, still doubt it will be less fragile than just using regex.

Reply:
Up to you, I approved if you don't think it's worth it

Reply:
I'll let @DonJayamanne take a look, since he actually owns the logging area